unit_tests/ngp_algorithms/UnitTestNodalGradPOpenBoundary.C

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#include "kernels/UnitTestKernelUtils.h"
+
+#include "UnitTestHelperObjects.h"
+
+#include "ngp_algorithms/UnitTestNgpAlgUtils.h"
+#include "ngp_algorithms/NodalGradPOpenBoundaryAlg.h"
+#include "ngp_algorithms/GeometryBoundaryAlg.h"
+#include "ngp_algorithms/NodalGradAlgDriver.h"
+#include "ngp_algorithms/GeometryAlgDriver.h"
+
+TEST_F(LowMachKernelHex8Mesh, NGP_nodal_grad_popen)
+{
+  // Only execute for 1 processor runs
+  if (bulk_.parallel_size() > 1) return;
+
+  const std::string meshSpec = "generated:20x20x20";
+  const bool doPerturb = false;
+  const bool generateSidesets = true;
+
+  fill_mesh_and_init_fields(meshSpec, doPerturb, generateSidesets);
+
+  unit_test_utils::HelperObjects helperObjs(bulk_, stk::topology::HEX_8, 1, partVec_[0]);
+
+  const double zero= 0.0;
+  const double one = 1.0;
+  const double two = 2.0;
+  const double oneVecTwelve[12] = {one, one, one, one, one, one, one, one, one, one, one, one};
+  stk::mesh::field_fill(zero, *dpdx_);
+  stk::mesh::field_fill(two, *dnvField_);
+  stk::mesh::field_fill(one, *pressure_);
+  stk::mesh::field_fill_component(oneVecTwelve, *exposedAreaVec_);
```

Comment:
`dnvField_` and `exposedAreaVec_` are computed using master elements within `fill_mesh_and_init_fields()`, what is the motivation for overriding these? And they get overwrriten by lines 43-48 anyway.

Reply:
Yes, it is not needed to initialize the dual nodal volume and exposed area vectors.